### PR TITLE
OAUTH: add --prompt option for Authentication Request

### PR DESCRIPTION
**Pull request by Slamdunk**
Close https://github.com/smallstep/cli/issues/215

### CODE REVIEW

**Comment by maraino** [comment]
Hi @Slamdunk, thanks for the PR, do you think we should validate the prompt arguments (none, login, consent, ...)? Do you know if they are standardized or are they IID-specific?

**Comment by Slamdunk** [comment]
So, from what I can tell, most things in the reference page https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest that have **not** the words MUST or REQUIRED are implemented slightly differently in each provider:

1. [Microsoft states](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow#request-an-authorization-code) it implements all 4, but `The only valid values at this time are login, none, and consent.`
2. [Google doesn't mention](https://developers.google.com/identity/protocols/oauth2/openid-connect#authenticationuriparameters) the `login` value
3. Neither [Github](https://docs.github.com/en/developers/apps/authorizing-oauth-apps) nor [Gitlab](https://docs.gitlab.com/ee/api/oauth2.html) have implemented anything about `prompt`

So to me the value shouldn't be validated for 2 reasons:

1. The RFC about `prompt` is merely a suggestion
2. It would be awkward to me to have `step-cli` telling `prompt=login` is valid, but then Google ignores it

If we don't provide any validation, `step-cli` users would be (a bit more) aware that they have to check their provider for the result, and to no expect `step-cli` to provide the ultimate truth about it.

File: command/oauth/cmd.go [hunk 1]
```diff
@@ -181,6 +181,10 @@ $ step oauth --client-id my-client-id --client-secret my-client-secret \
 				Name:  "scope",
 				Usage: "OAuth scopes",
 			},
+			cli.StringFlag{
+				Name:  "prompt",
+				Usage: "Whether the Authorization Server prompts the End-User for reauthentication and consent",
```
Comment (maraino) [inline command/oauth/cmd.go:186]: @Slamdunk I think we should add to the flag description what are the defined values, either just by name or even with a description formatting those like
https://github.com/smallstep/cli/blob/533e53e9ef691bc5c52cd548355b9b2c06eddaf1/command/crypto/jwk/create.go#L202-L215

**Review by maraino — COMMENTED** [review]
@Slamdunk Can you add the change in the description and I will merge.

File: command/oauth/cmd.go (hunk 1 above)
Comment (Slamdunk) [reply command/oauth/cmd.go]: Done:

image

**Review by maraino — APPROVED** [review]
Thanks again @Slamdunk, I've had some time again to look into this and it looks good.

Merging right away.

**Pull request merged**